src/core/shape_inference/include/experimental_detectron_prior_grid_generator_shape_inference.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/experimental_detectron_prior_grid_generator.hpp>
+
+namespace ov {
+namespace op {
+namespace v6 {
+
+template <class T>
+void shape_infer(const ExperimentalDetectronPriorGridGenerator* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 3 && output_shapes.size() == 1);
+    const auto& priors_shape = input_shapes[0];
+    const auto& featmap_shape = input_shapes[1];
+    const auto& im_data_shape = input_shapes[2];
+
+    auto& output_shape = output_shapes[0];
+    size_t output_size = op->m_attrs.flatten ? 2 : 4;
+
+    output_shape.resize(output_size);
+    output_shape[output_size - 1] = 4;
+
+    if (priors_shape.rank().is_dynamic() || featmap_shape.rank().is_dynamic()) {
+        return;
+    }
```

Comment:
To my we can't stop shape inference here in case when one of the ranks is dynamic. E. g. we can have dynamic priors_shape.rank(), but static featmap_shape.rank(). In this case we will infer output_shape[0] and output_shape[1] if op->m_attrs.flatten == false